test/dtypes/test_fp8.py

```diff
@@ -8,38 +8,55 @@
     run_tests,
 )
 from torchao.quantization.utils import TORCH_VERSION_AFTER_2_4
+
 try:
-    from torchao.prototype.fp8 import gemm_split_k
+    from torchao.prototype.fp8 import gemm_split_k, to_float8
     triton_available = True
 except ImportError:
     triton_available = False
 
+from torchao.utils import skip_if_compute_capability_less_than
+
 @unittest.skipIf(not triton_available, "Triton is required but not available")
 @unittest.skipIf(not torch.cuda.is_available(), "CUDA is required")
 class TestFP8Gemm(TestCase):
-    # @parametrize("dtype", [torch.bfloat16, torch.float16, torch.float32])
+    @skip_if_compute_capability_less_than(9.0)
     def test_gemm_split_k(self):
-        m, n, k = 256, 256, 512
+        dtype = torch.float16
+        qdtype = torch.float8_e4m3fn
 
-        a = torch.randn((m, k), dtype=torch.float16, device="cuda")
-        b = torch.randn((k, n), dtype=torch.float16, device="cuda")
-        c = gemm_split_k(a, b)
-        c_expected = torch.matmul(a, b)
-        assert torch.allclose(c, c_expected, atol=0.07) # less than this and the accuracy check fails
+        torch.cuda.manual_seed(0)
+
+        m = 64
+        n = 4096
+        k = 4096
+
+        # create test inputs
+        x = torch.randn((m, k), dtype=dtype, device='cuda')
+        w = torch.randn((n, k), dtype=dtype, device='cuda')
+
+        x_fp8, x_inv_s = to_float8(x, dtype=qdtype)
+        w_fp8, w_inv_s = to_float8(w, dtype=qdtype)
+
+        y_torch, _ = torch._scaled_mm(x_fp8, w_fp8.t(), out_dtype=dtype, scale_a=x_inv_s, scale_b=w_inv_s)
+        y_triton = gemm_split_k(x_fp8, w_fp8.t(), scale_a=x_inv_s.item(), scale_b=w_inv_s.item())
```

Comment:
These `.item` calls are probably annoying. I don't even think they're strictly necessary?

Reply:
Maybe just removing them I get this error

```
self = <triton.compiler.compiler.ASTSource object at 0x7fc290d0b2e0>
options = CUDAOptions(num_warps=8, num_ctas=1, num_stages=3, cluster_dims=(1, 1, 1), ptx_version=None, enable_fp_fusion=True, al...e', '/home/marksaroufim/.local/lib/python3.10/site-packages/triton/backends/nvidia/lib/libdevice.10.bc'),), debug=None)
codegen_fns = {'convert_custom_types': <function convert_custom_float8_sm90 at 0x7fc29901b520>}
context = <triton._C.libtriton.ir.context object at 0x7fc29bc39170>

    def make_ir(self, options, codegen_fns, context):
>       return ast_to_ttir(self.fn, self, context=context, options=options, codegen_fns=codegen_fns)
E       triton.compiler.errors.CompilationError: at 42:10:
E       
E               k_remaining = k - k_ * (block_k * split_k)
E       
E               a = tl.load(a_ptrs, mask=offs_k[None, :] < k_remaining, other=0.0)
E               b = tl.load(b_ptrs, mask=offs_k[:, None] < k_remaining, other=0.0)
E       
E               acc = tl.dot(a, b, acc, out_dtype=tl.float32)
E       
E               a_ptrs += block_k * split_k * stride_ak
E               b_ptrs += block_k * split_k * stride_bk
E       
E           acc = scale_a * scale_b * acc
E                 ^
E       IncompatibleTypeErrorImpl('invalid operands of type pointer<fp32> and pointer<fp32>')

../.local/lib/python3.10/site-packages/triton/compiler/compiler.py:112: CompilationError
```